src/averageAges.js

```diff
@@ -36,8 +43,23 @@ function calculateMenAverageAge(people, century) {
  *
  * @return {number}
  */
+
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const allWomen = people.filter(person => person.sex === 'f');
+
+  const mothers = people.map(person => person.mother);
+
+  const womenWithChildren = allWomen.filter(
+    woman => mothers.includes(woman.name)
+  );
+
+  const women = withChildren ? womenWithChildren : allWomen;
```

Comment:
ternaries should be multilined
```suggestion
  const women = withChildren 
    ? womenWithChildren 
    : allWomen;
```